modules/postgresql/main.tf

```diff
@@ -49,7 +49,7 @@ resource "google_sql_database_instance" "default" {
   provider            = google-beta
   project             = var.project_id
   name                = local.master_instance_name
-  database_version    = var.database_version
+  database_version    = (length(var.database_version) >= 8 && (substr(var.database_version, 0, 8) == "POSTGRES" || substr(var.database_version, 0, 8) == "postgres")) ? var.database_version : format("POSTGRES_%s", var.database_version)
```

Comment:
IIUC the logic is if it is number, prepend  `POSTGRES_` else pass var.database_version. Could we simplify the logic to something like `can(tonumber(var. database_version)) ? "POSTGRES_${var. database_version}" : var. database_version `. 

We could also add [variable validation](https://developer.hashicorp.com/terraform/language/values/variables#custom-validation-rules) rules to `var.database_version` to enforce one of number or string that starts with `POSTGRES_`.

Reply:
I will add variable validation rules, but "tonumber" function would not work if database version is 9_6 though it is a valid representation of database version, right?

Reply:
I have added variable validation rules for "database_version", which enforces that the version is of the pattern "POSTGRES_9_6" or "postgres_14" or "9_6".

After the validation, the main.tf file will check whether the database_version starts with a number, then it prepends it with "POSTGRES_" or returns the database version as it is.

Reply:
@ravisiddhu good point, we could do a replace for `_` but your new approach looks good.